### DISCUSSION TITLE
@react-native-firebase/ml-vision is deprecated how to do face detection now ?

### DISCUSSION THREAD

**Discussion by Daavidaviid** [invertase/react-native-firebase#4555] [Q&A] [answered]
Hi,

First, thanks for the great work.
I was wondering how are we supposed to do Face Detection **on the device** now that `@react-native-firebase/ml-vision` is deprecated ?

Because `@react-native-firebase/ml` is only for cloud ml.

**Comment by mikehardy** [comment, marked as answer]
That's exactly right and I have to beg your apologies for a little bit, as the replacement react-native-mlkit currently does not exist - I'm in the process of re-homing the functionality over there but until then there is *no* way to do it, *if* you upgrade

So the apologies are because for now, you are stuck on the last set of releases prior to #4471 / that release (that will not work) was rooted in the version of app 9.0.0, and the one prior that will work is 8.4.7 (from https://github.com/invertase/react-native-firebase/blob/master/packages/app/CHANGELOG.md) - along with the versions that require it as a peer dependency - all the ones that existed immediately prior to that PR merge and release on Nov 10

It is a very high priority item for me to get the react-native-mlkit package off the ground but unfortunately my high priority list is a bit deep right now. It might be a week or two

On the plus side, the older versions still should run fine, they were state of the art only 6 days ago :-)

**Reply by PhilipBrew** [reply]
Just wondering how this is going?

Thanks for the top work :)

**Reply by mikehardy** [reply]
Sorry - there is still no way to do it currently with react-native-firebase current stable, mlkit is still in the works, most of my other work queue is unwound now so this should be coming "soon", and I'm sorry for the delay - thanks for your patience

**Reply by thomas-blitzm** [reply]
Hi @mikehardy how is the MLKit going? I will be trying to use the older one for now. 

Will the new mlkit only be able to use URL for images or?

**Reply by mikehardy** [reply]
It's going poorly I must admit - I still have not had time. The combination of the pandemic causing me to be a homeschool teacher to support virtual school and my local school holiday schedule haven't let a lot of time. The general plan is for all of the code that existed here in react-native-firebase for local inference to move directly over to react-native-mlkit as a start and that's effectively it. Over time as the community supports it by proposing pull requests with any features that are not implemented by that code move, it will / could grow to cover all of google's mlkit.

**Reply by thomas-blitzm** [reply]
@mikehardy all good. No stress at all. 

I downgraded to older firebase as the main use of it is for react-native-camera and machine learning on barcodes. 

Thank you for all the hard work.

**Reply by StampixSMO** [reply]
Awesome work Mike, any updates on your roadmap?

**Comment by Daavidaviid** [comment]
I’d like to help if you want.

Em ter., 16/02/2021 às 20:01, Mike Hardy <notifications@github.com>
escreveu:

> It's going poorly I must admit - I still have not had time. The
> combination of the pandemic causing me to be a homeschool teacher to
> support virtual school and my local school holiday schedule haven't let a
> lot of time. The general plan is for all of the code that existed here in
> react-native-firebase for local inference to move directly over to
> react-native-mlkit as a start and that's effectively it. Over time as the
> community supports it by proposing pull requests with any features that are
> not implemented by that code move, it will / could grow to cover all of
> google's mlkit.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/invertase/react-native-firebase/discussions/4555#discussioncomment-375392>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHXUFFDR7KJZCYW3SD7GDH3S7L2NXANCNFSM4TXL4GNA>
> .
>
-- 
David Guerin
Software Engineer chez *Dev IT Go*
https://devitgo.com/

*Phone :* (+33) 7 57 90 53 46

**Reply by mikehardy** [reply]
I do actually really appreciate that, I'll commit to the first effort being just the pure infra work (directory skeleton, build system, github action -> npm publish) and opening it up so we can actually collaborate and start making it work! it's not like we're trying to be coy it's just empty at the moment. I'll schedule that for either late this week or early next. I do understand we have users currently left hanging and I'm not a fan of that either despite finite time. I do appreciate the understanding, thank you for that all

**Reply by Daavidaviid** [reply]
Ok cool no problem, I've been using this library so many times and I'd like to contribute for this part.

**Reply by Yonom** [reply]
I would also gladly help! Any updates on the repository?
